ffc-ahwr-event/index.mjs

```diff
@@ -5,27 +5,36 @@ import { saveMonitoringEvent } from './monitoring/protective-monitoring.mjs'
 import { onApplicationStatusEvent } from './application-status-event/index.mjs'
 import { onIneligibilityEvent } from './ineligibility-event/index.mjs'
 
-export default async function (context, message) {
-  const event = message
-  context.log.info(`Received event: ${JSON.stringify(event)}`)
+export default async function (context, event) {
+  const loggerInfo = {
+    name: event.name,
+    sbi: event.properties.sbi,
+    reference: event.properties.reference,
+    message: event.properties.action?.message
+  }
 
-  switch (event.name) {
-    case 'send-session-event':
-    case 'send-invalid-data-event':
-      if (validateEvent(event)) {
-        await saveEvent(context, event)
-      }
-      break
-    case 'application-status-event':
-      await onApplicationStatusEvent(context, event)
-      break
-    case 'send-ineligibility-event':
-      await onIneligibilityEvent(context, event)
-      break
-    default:
-      await saveMonitoring(context, event)
-      if (process.env.MONITORING_ENABLED === 'true') {
-        await saveMonitoringEvent(context, event)
-      }
+  try {
+    switch (event.name) {
+      case 'send-session-event':
+      case 'send-invalid-data-event':
+        if (validateEvent(event)) {
+          await saveEvent(context, event)
+        }
+        break
+      case 'application-status-event':
+        await onApplicationStatusEvent(context, event)
+        break
+      case 'send-ineligibility-event':
+        await onIneligibilityEvent(context, event)
+        break
+      default:
+        await saveMonitoring(context, event)
+        if (process.env.MONITORING_ENABLED === 'true') {
+          await saveMonitoringEvent(event)
+        }
+    }
+    context.log.info(`event received: ${JSON.stringify(loggerInfo)}`)
+  } catch (err) {
```

Comment:
If we catch error, then message will not be pushed back to DLQ as the function will complete as if no issue, and we'll never know the event was not processed

Reply:
good point, logging it and throwing again now.